### DISCUSSION TITLE
Swagger API documentation

### DISCUSSION THREAD

**Discussion by enrichman** [epinio/epinio#1195] [General]
While updating the API documentation that generates the swagger file I've seen that the models used to generate the doc are not the one that we are using as response in our APIs.

The main point of using annotations on the struct is to keep the documentation close to the code, so it can be easily maintained, and updated. Having the two splitted and unrelated adds more complexity to everything.

Do we want to refactor/fix this?

As an example I've looked ad the [ServiceAppsResponse](https://github.com/epinio/epinio/search?q=ServiceAppsResponse).

The response returned is [this](https://github.com/epinio/epinio/blob/7738a8164bf31df0c085c50d7108f5b3f91997aa/pkg/api/core/v1/models/models.go#L227-L230) one

```golang
// ServiceAppsResponse returns a list of apps per service
type ServiceAppsResponse struct {
	AppsOf map[string]AppList `json:"apps_of,omitempty"`
}
```

but the one that generates the doc is [this](https://github.com/epinio/epinio/blob/6fdba20bd4cfc4770a47d923d972a15cd6d90af5/internal/api/v1/docs/service.go#L28-L43) one

```golang
// swagger:route GET /namespaces/{Namespace}/serviceapps service ServiceApps
// Return map from services in the `Namespace`, to the apps in the same.
// responses:
//   200: ServiceAppsResponse

// swagger:parameters ServiceApps
type ServiceAppsParam struct {
	// in: path
	Namespace string
}

// swagger:response ServiceAppsResponse
type ServiceAppsResponse struct {
	// in: body
	Body map[string]models.AppList
}
```

There is no reference on the `apps_of` key, so its not reliable nor updated.

**Comment by andreas-kupries** [comment]
The 

> type ServiceAppsResponse struct {

etc. is an indirection swagger needs/uses to connect the comment about an API to the actual reponse and/or parameters.
The `Body` field is special there indicating the type of the body of the result.
That should be, for the example `models.ServiceAppsResponse`, i.e. the actual type used.

**Reply by enrichman** [reply]
Super, thanks for the explanation! We just need to remember to update them then! :D
